src/Microsoft.ApplicationInsights.AspNet/Extensibility/Implementation/Tracing/TelemetryLogger.cs

```diff
@@ -0,0 +1,120 @@
+        /// </summary>
+        public TelemetryLogger() : this(LogLevel.Error)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryLogger class.
+        /// </summary>
+        /// <param name="level">The logging level.</param>
+        public TelemetryLogger(LogLevel level)
+        {
+            this.Level = level;
+            this.eventSource = AspNetEventSource.Instance;
+
+            // The event levels are reordered for indexing by the log levels, this is a quick but potentially brittle way to do translation.
+            this.eventSourceLogCalls = new EventSourceLogCall[]
+            {
+                null,
+                this.eventSource.LogAlways,        // EventLevel.LogAlways = 0     | LogLevel.Debug = 1
+                this.eventSource.LogVerbose,       // EventLevel.Verbose = 5       | LogLevel.Verbose = 2
+                this.eventSource.LogInformational, // EventLevel.Informational = 4 | LogLevel.Information = 3
+                this.eventSource.LogWarning,       // EventLevel.Warning = 3       | LogLevel.Warning = 4
+                this.eventSource.LogError,         // EventLevel.Error = 2         | LogLevel.Error = 5
+                this.eventSource.LogCritical,      // EventLevel.Critical = 1      | LogLevel.Critical = 6
+            };
+        }
+
+        /// <summary>
+        /// Delegate of basic event source function calls.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        /// <param name="appDomainName">An ignored placeholder to make EventSource happy.</param>
+        private delegate void EventSourceLogCall(string message, string appDomainName = "Incorrect");
+
+        /// <summary>
+        /// Gets an instance of the telemetry logger.
+        /// </summary>
+        public static TelemetryLogger Instance
+        {
+            get
+            {
+                return TelemetryLogger.instance;
+            }
+        }
+
+        /// <summary>Gets or sets the currently configured logging level.</summary>
+        public LogLevel Level { get; set; }
+
+        /// <summary>
+        /// Begins a logical operation scope.
+        /// </summary>
+        /// <param name="state">The identifier for the scope.</param>
+        /// <returns>An IDisposable that ends the logical operation scope on dispose.</returns>
+        public IDisposable BeginScopeImpl(object state)
+        {
+            return new TelemetryLoggerScope();
+        }
+
+        /// <summary>
+        /// Checks if the given log level is enabled.
+        /// </summary>
+        /// <param name="logLevel">The leg level to check.</param>
+        /// <returns>True if the specified log level is enabled, otherwise false.</returns>
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel >= this.Level && this.Level != LogLevel.None && logLevel != LogLevel.None;
```

Comment:
The first check alone is not sufficient because "None" is defined as int.MaxValue instead of as 0 so it does not follow the sequence of the other values from Debug = 1 to Critical = 6.